cmd/storage-rest-server.go

```diff
@@ -285,6 +288,32 @@ func (s *storageRESTServer) ReadFileHandler(w http.ResponseWriter, r *http.Reque
 	w.Write(buf)
 }
 
+// ReadFileHandler - read section of a file.
+func (s *storageRESTServer) ReadFileStreamHandler(w http.ResponseWriter, r *http.Request) {
+	if !s.IsValid(w, r) {
+		return
+	}
+	vars := mux.Vars(r)
+	volume := vars[storageRESTVolume]
+	filePath := vars[storageRESTFilePath]
+	offset, err := strconv.Atoi(vars[storageRESTOffset])
+	if err != nil {
+		s.writeErrorResponse(w, err)
+		return
+	}
+	length, err := strconv.Atoi(vars[storageRESTLength])
+	if err != nil {
+		s.writeErrorResponse(w, err)
+		return
+	}
+	rc, err := s.storage.ReadFileStream(volume, filePath, int64(offset), int64(length))
+	if err != nil {
+		s.writeErrorResponse(w, err)
```

Comment:
already written error we should return right here.. we cannot proceed to do `io.Copy(w, rc)`